lib/core/Actions.ts

```diff
@@ -644,15 +685,18 @@ class Actions {
     return this.#request.post(action, data);
   }
 
-  #needsLogin(id) {
-    return [
-      'FElibrary',
-      'FEhistory',
-      'FEsubscriptions',
-      'SPaccount_notifications',
-      'SPaccount_privacy',
-      'SPtime_watched'
-    ].includes(id);
+  #needsLogin(id: string) {
+    switch (id) {
```

Comment:
I believe includes was more suitable here, a whole switch may be too verbose. What do you think?

Reply:
> The main reason was performance, although this might be an unnecessary optimization. Because with the switch it is (depending on the engine), probably just a lookup table, versus a throwaway allocation and loop over the elements of the array, on top of the `Object.is` equality check. 

I get that but it wouldn't make a lot of difference as far as I'm concerned. In this particular situation, a switch statement is probably overkill. Especially when all we're doing is checking whether an ordinary string exists in an array, which is what `includes()` was made for.

Reply:
> Also this does bring up the point, should I continue to make small changes like this to the code? This is just mainly moving types from here to there, but I think it would be beneficial to do stuff like this as I'm forced to look over every file anyways.

Sure.

Reply:
Okay, I'll revert the change.